File: BancoDev - BackEnd/MicroservicioEmpleado/src/main/java/com/bancoDev/Services/SucursalService.java
package com.bancoDev.Services;

import java.util.List;

import com.bancoDev.DTOs.ApiResponse;
import com.bancoDev.Models.SucursalEntity;

public interface SucursalService {

    ApiResponse<List<SucursalEntity>> listadoSucursales();    

}

File: bancoDev/src/app/services/sucursal.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environment/environment';
import { ApiResponse } from '../interfaces/ApiResponse';
import { SucursalEntity } from '../interfaces/empleado/SucursalEntity';

@Injectable({
  providedIn: 'root'
})
export class SucursalService {

  private api:string = `${environment.url}/sucursal`;

  constructor( private http: HttpClient ) { }

  listadoSucursales(): Observable<ApiResponse<SucursalEntity[]>> {
    return this.http.get<ApiResponse<SucursalEntity[]>>(`${this.api}/listar`)
  }

}
